hs_core/tests/api/native/test_core_metadata.py

```diff
@@ -698,80 +817,107 @@ def test_language(self):
-                                             url="http://hydroshare-1.org")
-            self.assertNotEqual(self.res.metadata.publisher.url, "http://hydroshare-1.org",
-                                msg="Resource publisher name matches.")
-
-            # publisher name nor the url can't be changed for a resource that has file(s) - should raise exception
-            self.assertRaises(Exception, lambda: resource.update_metadata_element(self.res.short_id, 'publisher',
-                                                                                  self.res.metadata.publisher.id,
-                                                                                  name="USGS", url="http://usgs.gov"))
-
-            # publisher element can't be deleted when the resource has public or published status - should raise exception
-            self.assertRaises(Exception, lambda:resource.delete_metadata_element(self.res.short_id, 'publisher',
-                                                                                 self.res.metadata.publisher.id))
-
-            # delete the file the resource has and then try to change the name of the publisher to anything you
-            # want - that should work
-            hydroshare.delete_resource_file(self.res.short_id, original_file_name, self.user)
-            resource.update_metadata_element(self.res.short_id,'publisher', self.res.metadata.publisher.id,
-                                             name="USGS", url="http://usgs.gov")
-            self.assertEqual(self.res.metadata.publisher.name, "USGS", msg="Resource publisher name did not match.")
-
-            # test that the publisher element can be deleted if the resource is private (not shared)
-            self.res.raccess.public = False
-            self.res.raccess.save()
-            resource.delete_metadata_element(self.res.short_id,'publisher', self.res.metadata.publisher.id)
+        # publisher element can't be added when the resource is not published
+        self.res.raccess.published = False
+        self.res.raccess.save()
+        self.assertRaises(Exception, lambda: resource.create_metadata_element(self.res.short_id, 'publisher',
+                                                                              name='HydroShare',
+                                                                              url="http://hydroshare.org"))
+
+        # publisher element can be added when the resource is published
+        self.res.raccess.published = True
+        self.res.raccess.save()
+        resource.create_metadata_element(self.res.short_id, 'publisher', name='USGS', url="http://usgs.gov")
+        self.assertEqual(self.res.metadata.publisher.url, "http://usgs.gov",
+                         msg="Resource publisher url did not match.")
+        self.assertEqual(self.res.metadata.publisher.name, "USGS", msg="Resource publisher name did not match.")
```

Comment:
these tests need to be updated if we chance Publisher element create() method to enforce publisher to be CUAHSI as I commented above.